z_infra_provisioning/cloud_infra_center/ocp_upi/README.md

```diff
@@ -80,53 +81,123 @@ Depending on the type of image registry backend an additional 100 GB volume.
-  - inventory: `bastion`, `vars.cluster_domain_name`, `vars.cluster_nodes`
+  - Ensure network connectivity among CIC control nodes, DNS, DHCP(if used),HAProxy, Installer or Bastion node.
+  - You can use your own Bastioin Node that installed DNS server and Load Balancer, or choose a RHEL8 machine.
+    - Ensure that you can ssh to bastion node or RHEL8 machine via root user without password.
+    - Uncomment `bastion` and `vars` part in inventory.yaml and fill the information of bastion node.
+    - inventory: `bastion.ansible_ssh_host`, `vars.cluster_domain_name`, `vars.os_dns_domain`.
+    - If using a new RHEL 8 machine as bastion node, please remember to run [Configure bastion](#configure-bastion) step to automatic create DNS and HAProxy.
 
-## Install Ansible
 
-This repository contains Ansible playbooks to deploy OpenShift on IBM® Cloud Infrastructure Center.
+## Getting Started
 
-**Requirements:**
+You can use any ICIC nodes or any RHEL8 machine to run ansible scripts to deploy OCP UPI, it's recommended to use RHEL8 machine. And the ansible script will do the following steps:
+* [Provision a Red Hat machine](#provision-a-red-hat-machine)
+* [Register the Red Hat machine](#register-the-red-hat-machine)
+* [Install required packages](#install-required-packages)
+* [Generate an SSH private key and add it to the CIC control node](#generate-an-ssh-private-key-and-add-it-to-the-cic-control-node)
+* [Connect the CIC client by using command line](#connect-the-cic-client-by-using-command-line)
+* [OpenShift Configuration Directory](#openshift-configuration-directory)
+* [Prepare the configuration before installation](#prepare-the-configuration-before-installation)
+* [Run ansible playbook to deploy OCP UPI](#run-ansible-playbook-to-deploy-ocp-upi)
 
-* Python
-* Ansible
-* Python modules required in the playbooks. Namely:
-  * openstackclient
-  * openstacksdk
-  * netaddr
+## Provision a Red Hat machine
+Provision a Red Hat machine based on the following specifications, or using existing ICIC compute nodes, the minmum system requirements: 1 vCPU, 2GB memory, 30GB Disk.
 
-### RHEL
+## Register the Red Hat machine 
 
-From a RHEL 8 box, make sure that the repository origins are all set:
+Verify the running ansible host has installed required packages, if do not have installed it, please do following steps.
+
+From a Red Hat machine, make sure that the repository origins are all set:
 
 ```sh
 sudo subscription-manager register # if not done already
+sudo subscription-manager list --available  # search for the pool ID $YOUR_POOLID
 sudo subscription-manager attach --pool=$YOUR_POOLID # if not done already
 sudo subscription-manager repos --disable=* # if not done already
 
 ## TODO replace repos to s390x repos
 sudo subscription-manager repos \
   --enable=rhel-8-for-s390x-baseos-rpms \
-  --enable=openstack-16-tools-for-rhel-8-s390x-rpms \
   --enable=ansible-2.8-for-rhel-8-s390x-rpms \
   --enable=rhel-8-for-s390x-appstream-rpms
 ```
```

Comment:
In my understanding, we haven't tested if these repos works. So we should delete this part? @zhengxiaomei123 Any comments.

Reply:
I tested those repo in new RedHat8 machine three times,  it should be worked in our 390x stand, but I have one problem for the openstackclient, the `pip3 install openstackclient` command will be failed because the `setup_tools` packages has some errors in s390x, but won't affect the installation.

Reply:
"Verify the running ansible host has installed required packages, if do not have installed it, please do following steps." 

should read 

"Verify the running Ansible host has installed the required packages, if they are not installed, please perform the following steps.

Reply:
Thanks, I've updated it.